src/viam/sdk/resource/resource_manager.hpp

```diff
@@ -1,33 +1,38 @@
 #pragma once
 
+#include <memory>
 #include <string>
 #include <unordered_map>
 
+#include <boost/optional/optional.hpp>
+#include <grpcpp/impl/service_type.h>
+
+#include <viam/api/component/generic/v1/generic.grpc.pb.h>
+
+#include <viam/sdk/resource/resource.hpp>
 #include <viam/sdk/resource/resource_base.hpp>
-#include <viam/sdk/resource/resource_type.hpp>
 
 namespace viam {
 namespace sdk {
 
 class ResourceManager {
    public:
-    void register_resource(std::shared_ptr<ResourceBase> Resource);
-
-    /// returns a resource from the registry.
-    /// Args:
-    /// 	std::string name: the name of the resource
-    ///
-    /// Raises:
-    /// 	If the name is not within the ResourceManager or the registered
-    ///     resource's type is not the expected type, then get_resource
-    ///     will throw an error.
-    std::shared_ptr<ResourceBase> get_resource(std::string name, ResourceType of_type);
-    ResourceManager();
+    std::shared_ptr<ResourceBase> resource(std::string name);
+    void replace_all(std::unordered_map<Name, std::shared_ptr<ResourceBase>> resources);
+    void add(Name name, std::shared_ptr<ResourceBase> resource);
+    void add(std::string name, std::shared_ptr<ResourceBase> resource);
+    void remove(Name name);
+    void replace_one(Name name, std::shared_ptr<ResourceBase> resource);
     const std::unordered_map<std::string, std::shared_ptr<ResourceBase>>& resources() const;
+    ResourceManager(){};
 
    private:
-    ResourceManager(std::vector<std::shared_ptr<ResourceBase>> resources);
+    std::mutex lock_;
     std::unordered_map<std::string, std::shared_ptr<ResourceBase>> resources_;
+    std::unordered_map<std::string, std::string> short_names_;
```

Comment:
[nit] Can you add a brief comment here explaining what a short name is

Reply:
Yep! Also added more documentation to `resource_manager.hpp` in general.